### ISSUE TITLE
dpkg-deb: error: 'rexray_0.9.2-1_amd64.deb' is not a debian format archive

### ISSUE THREAD

**Issue by carl-utter** [codedellemc/rexray#909]
Summary
Using Packer to provision AWS AMI instance:
`curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable`

## Expected Behavior
Action:
AMI with REXRAY included

## Actual Behavior
Getting error on packer build
"dpkg-deb: error: 'rexray_0.9.2-1_amd64.deb' is not a debian format archive"

## Steps To Reproduce
`curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable`

Full Packer File JSON is:
```
{
  "builders": [{
    "type": "amazon-ebs",
    "region": "us-east-1",
    "availability_zone": "us-east-1a",
    "source_ami_filter": {
      "filters": {
        "virtualization-type": "hvm",
        "name": "*ubuntu-xenial-16.04-amd64-server-*",
        "root-device-type": "ebs"
      },
      "owners": ["099720109477"],
      "most_recent": true
    },
    "instance_type": "t2.micro",
    "ssh_username": "ubuntu",
    "ami_name": "us-east-1-node",
    "force_deregister": true
  }],
  "provisioners": [{
    "type": "shell",
    "inline": [
      "sleep 15",
      "sudo apt-get clean",
      "sudo apt-get update",
      "sudo apt-get install -y apt-transport-https ca-certificates nfs-common",
      "curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -",
      "sudo add-apt-repository \"deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable\"",
      "sudo apt-get update",
      "sudo apt-get install -y docker-ce",
      "sudo usermod -aG docker ubuntu",
      "curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable"
    ]
  }, {
    "type": "file",
    "source": "docker.service",
    "destination": "/tmp/docker.service"
  }, {
    "type": "shell",
    "inline": [
      "sleep 15",
      "sudo ssh-keygen -b 4096 -f /home/ubuntu/.ssh/id_rsa -t rsa -N ''",
      "sudo chmod 400 /home/ubuntu/.ssh/id_rsa",
      "sudo chmod 700 /home/ubuntu/.ssh/id_rsa.pub",
      "sudo chown ubuntu:ubuntu /home/ubuntu/.ssh/id_rsa",
      "sudo chown ubuntu:ubuntu /home/ubuntu/.ssh/id_rsa.pub",
      "sudo mv /tmp/docker.service /lib/systemd/system/docker.service",
      "sudo chmod 644 /lib/systemd/system/docker.service",
      "sudo systemctl daemon-reload",
      "sudo systemctl restart docker",
      "sudo mkdir -p /workspace",
      "sudo chmod 777 /workspace",
      "sudo sysctl -w vm.max_map_count=262144"
    ]
  }]
}
```

**Comment by clintkitson** [comment]
@carl-utter Thanks for reporting this. Interesting, the `deb` packages are missing for 0.9.2. You can manually specify 0.9.1 with `curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable`. We will investigate further.

**Comment by akutz** [comment]
Travis-CI just updated all their Ubuntu worker images. I bet that broke the deb package install for the dpkg tools I use to build the pkgs. Ugh. Thanks for the notice.

-- 
-a

**Comment by akutz** [comment]
Yep, it was their damn upgrade. Broke fakeroot pkg. I'll get those debs up for 0.9.2 and fix this for future releases.

![img_8412](https://user-images.githubusercontent.com/101085/27715880-e2b6b76e-5cff-11e7-9009-825c2d4de623.PNG)
![img_8413](https://user-images.githubusercontent.com/101085/27715881-e2cf0f94-5cff-11e7-8549-b5f30d12d0f0.PNG)

**Comment by carl-utter** [comment]
@clintkitson the curl command you provided for manual specification to fetch the 0.9.1 version `curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable` is exactly the same as what I was using and which is attempting to fetch the 0.9.2 version.

What is the correct curl command format to use and be version specific for rexray?

**Comment by thomasbiddle** [comment]
Not working for myself either; definitely not Packer-specific as I've tried with both. Specifically specifying `0.9.1` still works - so just the 0.9.2 version.

**Comment by clintkitson** [comment]
@carl-utter I think my comment is exactly the same as your second one including 0.9.1? We have identified the issue as missing deb packages from the 0.9.2 release.

**Comment by daten-kieker** [comment]
We encountered the same problem. The workaround with pinning the version to 0.9.1 is working.

Any idea when debian packages will be back? Do we have to wait for TravisCI to fix the environment or is there something we can do?

**Comment by thomasbiddle** [comment]
@akutz Has this already been deployed? I'm still encountering the same issue after the #919 fix.

The issue is not that the `.deb` was not generated (Which seems like the confirmation in #919 is testing); but that it was generated with an invalid format, see below with a pull from the latest:

```
curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable

dpkg-deb: error: 'rexray_0.9.2-1_amd64.deb' is not a debian format archive
dpkg: error processing archive rexray_0.9.2-1_amd64.deb (--install):
 subprocess dpkg-deb --control returned error exit status 2
Errors were encountered while processing:
 rexray_0.9.2-1_amd64.deb
```

We can validate that that the package was generated correctly via `dpkg-deb --info file_name.deb` which is currently returning an exit-code of 2 with the following error:

```
dpkg-deb: error: 'rexray_0.9.2-1_amd64.deb' is not a debian format archive
```

**Comment by akutz** [comment]
That's because it's pulling down a nil stream. The fix didn't place a deb online for the missing version. I still have to generate that.

**Comment by akutz** [comment]
Hi @thomasbiddle,

I've [uploaded](https://emccode.bintray.com/rexray/stable/0.9.2/) the DEB packages for 0.9.2. Future deployments will work automatically with the `fakeroot` fix from #919. Thank you again for pointing this out!

![image](https://user-images.githubusercontent.com/101085/27927904-d7e7c636-6252-11e7-8f35-37c0b25c9af0.png)
